clover/netcdf/variable.py

```diff
@@ -464,6 +464,35 @@ def datetimes(self):
         else:
             return numpy.array([datetime(*d.timetuple()[:6], tzinfo=pytz.UTC) for d in self.dates])
 
+    @property
+    def unit(self):
+        def varys_by_year(x, y):
+            if y.year == x.year or (y - x).seconds != 0 or x.month != y.month or x.day != y.day:

```

Comment:
why not simplify a little?  This seems like it would work:

```
if y.year == x.year or not (y-x):
```
From the docs `in Boolean contexts, a timedelta object is considered to be true if and only if it isn’t equal to timedelta(0).`